rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake

```diff
@@ -122,7 +122,7 @@ list(APPEND _generated_msg_headers "${_visibility_control_file}")
 
 set(_target_suffix "__rosidl_generator_c")
 
-add_library(${rosidl_generate_interfaces_TARGET}${_target_suffix} SHARED
+add_library(${rosidl_generate_interfaces_TARGET}${_target_suffix} ${rosidl_generator_c_LIBRARY_TYPE}
```

Comment:
Since this happens outside of this package, maybe worth comparing `rosidl_generator_c_LIBRARY_TYPE` with `BUILD_SHARED_LIBS`? Unless you can think of a scenario where that's desired (I cannot), then it's almost certainly a mistake in the build setup.

Reply:
`BUILD_SHARED_LIBS` might not even be defined in a package which runs this CMake code. Also I wouldn't restrict the combination here. It sounds viable to me that a package builds itself with one library type while the typesupport package was built with a different one (so do the message specific libraries created by this CMake snippet).

Reply:
> BUILD_SHARED_LIBS might not even be defined in a package which runs this CMake code. 

`if(DEFINED BUILD_SHARED_LIBS AND ...`?

> It sounds viable to me that a package builds itself with one library type while the typesupport package was built with a different one (so do the message specific libraries created by this CMake snippet).

I suppose that's the case, it's probably not worth doing then.